engine/render/src/render/font_renderer.cpp

```diff
@@ -562,6 +563,36 @@ namespace dmRender
         te.m_SourceBlendFactor = params.m_SourceBlendFactor;
         te.m_DestinationBlendFactor = params.m_DestinationBlendFactor;
 
+        TextMetrics metrics;
+        uint32_t line_count = GetTextMetrics(font_map, params.m_Text, params.m_Width, params.m_LineBreak, params.m_Leading, params.m_Tracking, &metrics);
+        float x_offset = OffsetX(te.m_Align, te.m_Width);
+        float y_offset = OffsetY(te.m_VAlign, te.m_Height, font_map->m_MaxAscent, font_map->m_MaxDescent, te.m_Leading, line_count);
+
+        // find X,Y local coordinate of text center
+        float center_x = x_offset; // start from the X position of the pivot point
+        switch (te.m_Align) {
+            case TEXT_ALIGN_LEFT:
+                center_x += metrics.m_Width/2; // move halfway to the right since we're aligning left
+            break;
+            case TEXT_ALIGN_RIGHT:
+                center_x -= metrics.m_Height/2; // move halfway to the left from pivot since we're aligning right
+            break;
+            // nothing to do for TEXT_ALIGN_CENTER. Pivot is already at the center of the text X-wise
+        }
+        float center_y = y_offset + font_map->m_MaxAscent - metrics.m_Height/2; // 'y_offset' to move to the baseline of first letter in text. +'MaAscent' to get to the top of the text. -layout_height to get to the center.
+
+        // find center and radius for frustum culling
+        dmVMath::Vector4 centerpoint_local(center_x, center_y, 0, 1);
+        dmVMath::Vector4 cornerpoint_local(centerpoint_local.getX() + metrics.m_Width/2, centerpoint_local.getY() + metrics.m_Height/2, centerpoint_local.getZ(), centerpoint_local.getW());
```

Comment:
I recommend doing it in one statement:
```
dmVMath::Point3(center_x + metrics.m_Width/2, center_y + metrics.m_Height/2, 0);
```

The dmVMath::Point3() represents a point, i.e it sets `w=1`.
Using the correct type helps when using the math library, since it have make decisions base on if it's a point or not.

Reply:
@JCash , note there are _two_ different points there. 

There is one for center (centerpoint_local) and another for the corner (**corner**point_local). The second is placed in relation to the first. I've used `centerpoint_local.getX() + ...` to emphasize the fact of this relation. `getX()` is also inlined so i think there is no performance hit for it.

Does your comment still apply ?

Reply:
No, I was simply misled by the very similar variable names.